mantle/platform/api/aliyun/api.go

```diff
@@ -128,14 +129,51 @@ func (a *API) CopyImage(source_id, dest_name, dest_region, dest_description, kms
 			Value: "mantle",
 		},
 	}
-
 	response, err := a.ecs.CopyImage(request)
 	if err != nil {
 		return "", fmt.Errorf("copying image: %v", err)
 	}
+
+	// if we have a need to operate on an image immediately after the image has
+	// been copied to a region, we can wait for it to be marked available
+	//
+	// NB: this gem from the Aliyun API docs - "A single region can have only
+	// one image copy task running at a time. Other image copy tasks queue up
+	// for the current task to complete before they run in sequence."
+	//
+	// Not sure if this means only one copy task can run for the *entire* region
+	// or if the limitation is per account, per region...but this kind of
+	// queuing would explain some of the delays observed.
+	if wait_for_ready {
+		plog.Infof("waiting for %v in %v to be available before returning", response.ImageId, dest_region)
+		a.WaitForImageReady(dest_region, response.ImageId)
+	}
 	return response.ImageId, nil
 }
 
+// WaitForImageReady checks that an image in a region is available to be
+// operated on. i.e. when you want to modify attributes of an image
+func (a *API) WaitForImageReady(region_id string, image_id string) error {
+	checkAvailable := func() error {
+		images, err := a.GetImagesByID(image_id, region_id)
+		if err != nil {
+			return fmt.Errorf("getting images: %v", err)
+		}
+		for _, img := range images.Images.Image {
+			if img.ImageId == image_id && img.Status == "Available" {
+				return nil
+			}
+		}
+		return fmt.Errorf("%v in %v was not available", image_id, region_id)
+	}
+
+	if err := util.RetryUntilTimeout(10*time.Minute, 10*time.Second, checkAvailable); err != nil {
```

Comment:
is 10 minutes long enough wait time?

Reply:
Worst case scenario is that every region (25) takes 10 minutes to succeed or fail, so that's looking at 4+ hrs we could spend trying to replicate. The Aliyun stage of the pipeline would timeout well before that, so I think 10 mins is fine.

We'd re-run the pipeline about 2-4 hrs after that kind of failure, so I don't think we should be waiting around too long for success/failure.